Client/Ecosia/UI/NTP/ClimateImpactCounter/SeedProgressManager.swift

```diff
@@ -0,0 +1,124 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol SeedProgressManagerProtocol {
+    static var progressUpdatedNotification: Notification.Name { get }
+    static func loadCurrentLevel() -> Int
+    static func loadTotalSeedsCollected() -> Int
+    static func loadLastAppOpenDate() -> Date
+    
+    static func saveProgress(totalSeeds: Int, currentLevel: Int, lastAppOpenDate: Date)
+    
+    static func addSeeds(_ count: Int)
+    static func resetCounter()
+    
+    static func calculateInnerProgress() -> CGFloat
+    static func collectSeed()
+}
+
+final class UserDefaultsSeedProgressManager: SeedProgressManagerProtocol {    
+    
+    static var progressUpdatedNotification: Notification.Name { .init("SeedProgressUpdated") }
+    private static let numberOfSeedsAtStart = 1
+    
+    // UserDefaults keys
+    private static let totalSeedsCollectedKey = "TotalSeedsCollected"
+    private static let currentLevelKey = "CurrentLevel"
+    private static let lastAppOpenDateKey = "LastAppOpenDate"
+    
+    // Thresholds for each level
+    private static let levelThresholds: [Int] = [5, 10] // Example thresholds for levels 1, 2
```

Comment:
Is the maximum seeds for the experiment going to be handled separately? I guess there's even a slightly different gray UI for that case.

Reply:
I added everything there at the moment. That's still quite vague but will double check and update if so 💪